### DISCUSSION TITLE
How to change  MainPage in Prism Navigation at run time

### DISCUSSION THREAD

**Discussion by sunilsolanki-alight** [PrismLibrary/Prism#2970] [.NET MAUI]
How to change  MainPage in Prism Navigation at run time, Also is there any alternate to get the navigationService.GetNavigationUriPath(); ?

**Comment by dansiegel** [comment]
Use an Absolute URI. This will reset the MainPage of the Window (formerly in Xamarin.Forms the Application.MainPage)

```cs
navigation.NavigateAsync("/AnotherPage");
```

**Reply by sunilsolanki-alight** [reply]
it reset complete window which give behavior like new app launch with splash screen.

**Reply by dansiegel** [reply]
You are clearly using the old preview on NuGet. This is fixed in .NET 8 with Prism 9

**Reply by sunilsolanki-alight** [reply]
Thanks @dansiegel , we using below version with .NET 7 
![image](https://github.com/PrismLibrary/Prism/assets/119950546/ac2a6016-22af-4f4e-affe-f615dac8390f)
Is it correct or need to install some other package.  

Hope once will update  to .NET 8 it works
